pytorch_lightning/plugins/io/xla_plugin.py

```diff
@@ -36,8 +37,18 @@ def save_checkpoint(self, checkpoint: Dict[str, Any], path: _PATH, storage_optio
         Args:
             checkpoint: dict containing model and trainer state
             path: write-target path
-            storage_options: Optional parameters when saving the model/training states.
+            storage_options: not used in ``XLACheckpointIO.save_checkpoint``
+
+        Raises:
+            MisconfigurationException:
+                If ``storage_options`` arg is passed in
         """
+        if storage_options is not None:
+            raise MisconfigurationException(
+                "`Trainer.save_checkpoint(..., storage_options=...)` with `storage_options` arg "
+                f"is not supported for `{self.__class__.__name__}`. Please implement your custom `CheckpointIO`"
+                "to define how you'd like to use `storage_options`."
```

Comment:
```suggestion
                " to define how you'd like to use `storage_options`."
```
space is missing, in other messages too.
best is to put it in front so it doesn't get missed.

make sure to trigger the error at least once to inspect it :)) don't trust `pytest.raises`! 😄